crates/ibc/src/mock/host.rs

```diff
@@ -38,6 +39,7 @@ pub enum HostType {
 #[derive(Clone, Debug, PartialEq, Eq)]
 pub struct SyntheticTmBlock {
     pub trusted_height: Height,
+    pub trusted_next_validators: ValidatorSet,
```

Comment:
Very good question. To answer your question, no. (Light-)Blocks do not store `trusted_height` and `trusted_next_validators`. They are computed by the relayer by querying the remote blockchain. Actually, I will suggest removing these two fields from `SyntheticTmBlock` as these two fields are not required for a host context. These fields are only needed when the relayer submits a client update message to the counterparty chain.

Previously, in our Mock version, `trusted_next_validators` was computed as the next validator of the current header - which is semantically wrong. But works for consecutive client height updates.

And the same was being done in the issue reported by Nomic. This is why my previously written test should not have failed. But I made a mistake :sweat_smile: The fix is in a2163f1cd2b4f848ae285ec26bb9f0ac0719d7b1. With this fix, the bug couldn't have been caught by tests.

Note, I can't set `light_block.header.next_validator` later - as this will break the header signature verification.

Reply:
Let's tackle this in future PR or directly by mock abstraction - as this includes the existing `trusted_height` field.